src/js/redux/reducers.js

```diff
@@ -123,21 +123,76 @@ export function onPageObjectId(state, { payload }) {
 }
 
 export function onRegisterModule(state, { payload }) {
-  const isModuleLoaded = (state.modules || []).find((module) => Object.keys(module).find((key) => key === payload?.module));
+  const isModuleLoaded = state.modules[payload.module];
+  if (!isModuleLoaded) {
+    return {
+      ...state,
+      modules: {
+        ...state.modules,
+        [payload.module]: {
+          manifestLocation: payload.manifestLocation || payload.manifest,
+        },
+      },
+    };
+  }
+
+  return state;
+}
+
+export function loadNavigationReducer(state, { payload }) {
+  return {
+    ...state,
+    navigation: payload,
+  };
+}
+
+export function loadNavigationLandingPageReducer(state, { payload }) {
+  return {
+    ...state,
+    navigation: {
+      ...state.navigation,
+      landingPage: payload,
+    },
+  };
+}
+
+export function loadNavigationSegmentReducer(state, { payload: { segment, schema } }) {
+  return {
+    ...state,
+    navigation: {
+      ...state.navigation,
+      [segment]: schema,
+    },
+  };
+}
+
+export function loadModulesSchemaReducer(state, { payload: { schema } }) {
+  const scalprumConfig = Object.entries(schema).reduce(
+    (acc, [name, config]) => ({
+      ...acc,
+      [name]: {
+        name,
+        module: `${name}#./RootApp`,
+        manifestLocation: `${window.location.origin}${isBeta() ? '/beta' : ''}${config.manifestLocation}`,
```

Comment:
I think we could have precalculated `manifestLocation` from `name` if it's missing from config.

Reply:
I would like to avoid any calculation if possible. The module definition is not something that we will be creating/updating every few minutes. It will stay the same probably once an application is deployed for the first time. If the location is strictly set, we don't have to fear any potential edge cases in the future. No calculation equals no bugs.